devices.py

```diff
@@ -59,35 +61,296 @@ class Newport(EpicsMotor):
+    set_run_count_pattern = Cpt(EpicsSignal, ":PLYMOD")
+    set_run_count = Cpt(EpicsSignal, ":REPCNT")
+
+    # top row
+    sequence_owner = Cpt(EpicsSignal, ":HUTCH_ID")
+    sequence_owner_name = Cpt(EpicsSignal, ":HUTCH_NAME")
+    photon_beam_owner = FCpt(
+        EpicsSignal, "{self._beam_owner_prefix}:BEAM_OWNER_ID")
+    photon_beam_owner_name = FCpt(
+        EpicsSignal, "{self._beam_owner_prefix}:BEAM_OWNER_NAME")
     
+    # center column
+    beam_rate = FCpt(EpicsSignalRO, "{self._beam_rate_pv}")
+    play_count = Cpt(EpicsSignalRO, ":PLYCNT")
+    total_play_count = Cpt(EpicsSignalRO, ":TPLCNT")
+    play_status = Cpt(EpicsSignalRO, ":PLSTAT")
+    current_step = Cpt(EpicsSignalRO, ":CURSTP")
+
+    # right column
+    sync_marker = Cpt(EpicsSignal,":SYNCMARKER")
+    next_sync = Cpt(EpicsSignal,":SYNCNEXTTICK")
+    run_using = Cpt(EpicsSignal,":BEAMPULSEREQ")
+
+    # add some properties 
+    # learn about hints methods 
+
     def __init__(self, prefix, timeout=1, *args, **kwargs):
+        self._beam_rate_pv = "EVNT:SYS0:1:LCLSBEAMRATE"
+        self._beam_owner_prefix = "ECS:SYS0:0"
         super().__init__(prefix, *args, **kwargs)
         self.timeout = timeout
 
-    def start(self):
+    def start(self,wait=True):
         """
         Start the sequencer.
         """
         status = self.state_control.set(1, timeout=self.timeout)
-        status_wait(status)
+        if wait:
+            status_wait(status)
         
-    def stop(self):
+    def stop(self,wait=True):
         """
         Stop the sequencer.
         """
         status = self.state_control.set(0, timeout=self.timeout)
-        status_wait(status)        
+        if wait:
+            status_wait(status)        
+
+
+class ConsolidatedSamplePalette(Device):
+    '''
+    
+    '''
+    # x_motor = Cpt(IMS,"{self._xmotor_prefix}:{self._xmotor_suffix}")
+    # y_motor = Cpt(IMS,"{self._ymotor_prefix}:{self._ymotor_suffix}")
+    # z_motor = Cpt(IMS,"{self._zmotor_prefix}:{self._zmotor_suffix}")
+
+
+    def __init__(self, prefix, N_dim=1, M_dim=1, timeout=1, motor_timeout=5,
+                samples=None, *args, **kwargs):
+        '''
+        N_dim : int
+            specify the number of samples in the N direction
+
+        M_dim : int
+            specify the number of samples in the M direction
+        '''
+        
+        self._xmotor_prefix = None
+        self._ymotor_prefix = None
+        self._zmotor_prefix = None
+        self._xmotor_suffix = None
+        self._ymotor_suffix = None        
+        self._zmotor_suffix = None       
+
+        super().__init__(prefix, *args, **kwargs)
+        self.timeout = timeout
+        self.motor_timeout = motor_timeout
+
+        # How many samples are there in the N and M directions
+        self.N_dim = N_dim
+        self.M_dim = M_dim
+
+        # if the number of samples on the pallete is not given, presume full
+        # matrix allotment 
+        if samples == None:
```

Comment:
PEP8. In general, when checking if a variable has a value, best practice is to do:
```
if samples:
    ...
```
Or add a `not` in there if you need the negative. If you explicitly need to check if the value is `None`, then make the comparison as follows:
```
if samples is None:
    ...
```